src/SprykerEco/Zed/Computop/Business/Hook/ComputopPostSaveHook.php

```diff
@@ -55,24 +55,27 @@ public function registerMapper(InitMapperInterface $paymentMethod)
      */
     public function execute(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponseTransfer)
     {
+        $payment = $quoteTransfer->getPayment();
+
+        if ($payment->getPaymentProvider() !== ConputopSharedConfig::PROVIDER_NAME) {
+            return $checkoutResponseTransfer;
+        }
+
         $quoteTransfer->setOrderReference($checkoutResponseTransfer->getSaveOrder()->getOrderReference());
+        /** @var \Generated\Shared\Transfer\ComputopCreditCardPaymentTransfer $computopPaymentTransfer */
         $computopPaymentTransfer = $this->getPaymentTransfer($quoteTransfer);
 
-        if (
-            $quoteTransfer->getPayment()->getPaymentSelection() !== ConputopSharedConfig::PAYMENT_METHOD_PAY_NOW
-            && $quoteTransfer->getPayment()->getPaymentSelection() !== ConputopSharedConfig::PAYMENT_METHOD_EASY_CREDIT
-            && $quoteTransfer->getPayment()->getPaymentSelection() !== ConputopSharedConfig::PAYMENT_METHOD_CREDIT_CARD
-        ) {
-            return $this->setRedirect($computopPaymentTransfer, $checkoutResponseTransfer);
-        }
+        if (in_array($payment->getPaymentSelection(), $this->config->getPaymentMethodsWithInitPaymentTransfer())) {
+            $checkoutResponseTransfer->setComputopInitPayment(
```

Comment:
* Than do `setRedirect()`.
* In opposite case return `ComputopInitPaymentTransfer` as it was before.

Reply:
++